examples/benchmark_preference_tasks.py

```diff
@@ -0,0 +1,167 @@
+        B = example["conversation_b"][3]["content"]
+        prompt = example["conversation_a"][2]["content"]
+        generation = [A, B]
+    # dataset has label "tie", that will have label=2
+    gold = int(example["winner"] == "model_b") if example["winner"] != "tie" else 2
+    example["input"] = prompt
+    example["generations"] = generation
+    example["gold"] = gold
+    return example
+
+
+def format_columns_hhh(examples: Dataset) -> Dataset:
+    """
+    Format columns for the HHH dataset.
+
+    Args:
+        examples (Dataset): Input dataset.
+
+    Returns:
+        Dataset: Formatted dataset.
+    """
+    generations = []
+    gold_labels = []
+    for target in examples["targets"]:
+        generations.append(target["choices"])
+        # Conventional multiple-choice accuracy can be achieved by assigning the correct target
+        # a score of 1, and all incorrect targets a score of 0.
+        gold_labels.append(target["labels"].index(1))
+    examples["generations"] = generations
+    examples["gold"] = gold_labels
+    return examples
+
+
+def get_dataset(dataset_name: str) -> Tuple[Dataset, List[int]]:
+    """
+    Load and format the specified dataset.
+
+    Args:
+        dataset_name (str): Name of the dataset.
+
+    Returns:
+        Tuple[Dataset, List[int]]: Formatted dataset and gold labels.
+    """
+    if dataset_name == "MT_Bench":
+        dataset = load_dataset("lmsys/mt_bench_human_judgments", split="human")
+        dataset = dataset.select(random.sample(range(len(dataset)), 100))       
+        dataset = dataset.map(format_columns_lmsys)
+        gold = dataset["gold"]
+        cols_to_remove = ["question_id", "model_a", "model_b", "winner", "judge", "turn", "conversation_a", "conversation_b"]
+        dataset = dataset.remove_columns(cols_to_remove)
+        dataset = dataset.select(range(100))
+    elif dataset_name == "HHH_Alignment":
+        dataset = load_dataset("HuggingFaceH4/hhh_alignment", "other", split="test")
+        dataset = dataset.map(format_columns_hhh, batched=True)
+        gold = dataset["gold"]
+        dataset = dataset.remove_columns(["targets", "gold"])
+    else:
+        raise KeyError("Choose either MT_Bench or HHH_Alignment dataset")
+    return dataset, gold
+
+
+def generate_preds(dataset: Dataset, task_type: Task) -> Dataset:
+    """
+    Generate predictions using the specified task and dataset.
+
+    Args:
+        dataset (Dataset): Input dataset.
+        task_type (Task): Type of task.
+
+    Returns:
+        Dataset: Dataset with generated predictions.
+    """
+    generator = OpenAILLM(
+        task=task, 
+        prompt_format="openai",
+        max_new_tokens=1024,
+        openai_api_key="sk-YOUR-KEY",
+        temperature = 1.0,
+        top_p = 1.0,
+    )
+    pipe = pipeline("preference", "text-quality", labeller=generator)
+    dataset = pipe.generate(dataset, num_generations=1, display_progress_bar=True)
+    return dataset
```

Comment:
Maybe allow the OpenAI SH key to be passed as argument, so it can be introduced through main in case we need a rerun. I'd also allow this key to be passed as CLI parameter.